src/network/networkd-manager-varlink.c

```diff
@@ -259,6 +267,64 @@ static int vl_method_set_persistent_storage(sd_varlink *vlink, sd_json_variant *
         return sd_varlink_reply(vlink, NULL);
 }
 
+static int vl_method_set_link_up_or_down(sd_varlink *vlink, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata, bool up) {
+        Manager *manager = ASSERT_PTR(userdata);
+        Link *link = NULL;
+        int r;
+
+        assert(vlink);
+
+        r = dispatch_interface(vlink, parameters, manager, /* use_polkit = */ true, &link);
+        if (r != 0)
+                return r;
+
+        /* Require a specific link to be specified. */
+        if (!link)
+                return sd_varlink_error_invalid_parameter_name(vlink, "InterfaceIndex");
+
+        r = varlink_verify_polkit_async(
+                        vlink,
+                        manager->bus,
+                        "org.freedesktop.network1.manage-links",
+                        /* details= */ NULL,
+                        &manager->polkit_registry);
+        if (r <= 0)
+                return r;
+
+        /* Check if operation is redundant - link already in desired state */
+        bool current_up = FLAGS_SET(link->flags, IFF_UP);
+        if (current_up == up)
+                return sd_varlink_reply(vlink, NULL);
+
+        /* Check if there's already a pending set flags operation */
+        if (link->set_flags_messages > 0)
+                return sd_varlink_error_errno(vlink, EBUSY);
+
+        /* Reserve the operation slot immediately to prevent races */
+        link->set_flags_messages++;
```

Comment:
Hmm, I cannot follow this. Please elaborate more why this is necessary.

Basically, such 'workaround' to prevent race should be fixed in a different way.